KIPs/kip-37.md

```diff
@@ -0,0 +1,746 @@
+---
+kip: 37
+title: Token Standard
+author: Kyungkoo Kai Kim <kai.kim@groundx.xyz>
+discussions-to: https://github.com/klaytn/kips/issues/37
+status: Draft
+type: Standards Track
+category: KCT
+created: 2020-08-19
+requires: 13
+---
+
+## Simple Summary
+
+A standard interface for contracts that manage multiple token types. A single deployed contract may include any combination of fungible tokens, non-fungible tokens or other configurations (e.g. semi-fungible tokens).
+
+## Abstract
+
+This standard outlines a smart contract interface that can represent any number of fungible and non-fungible token types. Existing standards such as KIP-7 require deployment of separate contracts per token type. The KIP-17 standard’s token ID is a single non-fungible index and the group of these non-fungibles is deployed as a single contract with settings for the entire collection. In contrast, the KIP-37 Multi Token Standard allows for each token ID to represent a new configurable token type, which may have its own metadata, supply and other attributes.
+
+The `_id` argument contained in each function’s argument set indicates a specific token or token type in a transaction.
+
+## Motivation
+
+Tokens standards like KIP-7 and KIP-17 require a separate contract to be deployed for each token type or collection. This places a lot of redundant bytecode on the Klaytn blockchain and limits certain functionality by the nature of separating each token contract into its own permissioned address. With the rise of blockchain games and platforms like Enjin Coin, game developers may be creating thousands of token types, and a new type of token standard is needed to support them. However, KIP-7 is not specific to games and many other applications can benefit from this flexibility.
```

Comment:
```suggestion
Tokens standards like KIP-7 and KIP-17 require a separate contract to be deployed for each token type or collection. This places a lot of redundant bytecode on the Klaytn blockchain and limits certain functionality by the nature of separating each token contract into its own permissioned address. With the rise of blockchain games and platforms like Enjin Coin, game developers may be creating thousands of token types, and a new type of token standard is needed to support them. However, KIP-37 is not specific to games and many other applications can benefit from this flexibility.
```

Reply:
"KIP-7" in the last sentence should be "KIP-37", right?

Reply:
That's rignt.